provider/pkg/gen/go-templates/yaml/configFile.tmpl

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// *** WARNING: this file was generated by pulumigen. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+package yaml
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+// ConfigFile creates a set of Kubernetes resources from a Kubernetes YAML file.
+type ConfigFile struct {
+	pulumi.ResourceState
+
+	Resources map[string]pulumi.Resource
```

Comment:
I'm a little nervous that we are making this (and `GetResource`) prompt - since its a diverergence from all other langauges and a breaking change to make it align with other languages in the public API in the future.  

An option to make this public type match other languages, and still avoid all the internal complexity for now but just wrap into an `Output` before storing.

That said, this leads to a meaningfully simpler API, so I am open to just accepting the divergence if you feel that is best here.

Reply:
Hmm, yeah. I'm torn on this as well. The API is much nicer this way, but maybe we should stay consistent.

Perhaps we could offer both APIs?

Reply:
We decided to stick with the simple API for now. I opened https://github.com/pulumi/pulumi-kubernetes/issues/1108 to track further work on standardizing the SDKs.